mbed-client/m2mbase.h

```diff
@@ -469,17 +469,22 @@ protected : // from M2MReportObserver
 private:
 
     M2MReportHandler           *_report_handler;
-    M2MObservationHandler      *_observation_handler;
+    M2MObservationHandler      *_observation_handler;    
+    uint8_t                     *_resource_type;
+    uint8_t                     *_uri_path;
+    //uint8_t                     *_name;
     String                      _name;
-    String                      _resource_type;
-    String                      _interface_description;
-    String                      _uri_path;
+    uint8_t                      *_interface_description;
     int32_t                     _name_id;
     uint32_t                    _max_age;
     uint16_t                    _instance_id;
     uint16_t                    _observation_number;
     uint8_t                     *_token;
     uint8_t                     _token_length;
+    uint8_t                     _resource_type_length;
+    uint8_t                     _uri_path_length;
+    uint8_t                     _interface_description_length;
```

Comment:
Previous API did not have practical limit on these lengths. If you are using zero terminated strings, why still keep separate length field? The code is not too strict on overflow checks on length, as it has been 32bit.